autosklearn/pipeline/components/data_preprocessing/text_encoding/bag_of_word_encoding_distinct.py

```diff
@@ -0,0 +1,114 @@
+from autosklearn.pipeline.components.base import AutoSklearnPreprocessingAlgorithm
+from autosklearn.pipeline.constants import DENSE, SPARSE, UNSIGNED_DATA, INPUT
+
+from sklearn.feature_extraction.text import CountVectorizer
+
+
+class BagOfWordEncoder(AutoSklearnPreprocessingAlgorithm):
+    def __init__(
+        self,
+        ngram_range: Optional[int] = None,
+        min_df_choice: Optional[str] = None,
+        min_df_absolute: Optional[int] = None,
+        min_df_relative: Optional[float] = None,
+        random_state: Optional[Union[int, np.random.RandomState]] = None
+    ) -> None:
+        self.ngram_range = ngram_range
+        self.random_state = random_state
+        self.min_df_choice = min_df_choice
+        self.min_df_absolute = min_df_absolute
+        self.min_df_relative = min_df_relative
+
+    def fit(self, X: PIPELINE_DATA_DTYPE, y: Optional[PIPELINE_DATA_DTYPE] = None
+            ) -> 'BagOfWordEncoder':
+
+        if isinstance(X, pd.DataFrame):
+            # define a CountVectorizer for every feature (implicitly defined by order of columns,
+            # maybe change the list
+            # to a dictionary with features as keys)
+            if self.min_df_choice == "min_df_absolute":
+                self.preprocessor = [CountVectorizer(min_df=self.min_df_absolute,
+                                                     ngram_range=(1,
+                                                                  self.ngram_range)).fit(X[feature])
+                                     for feature in
+                                     X.columns]
+            elif self.min_df_choice == "min_df_relative":
+                self.preprocessor = [CountVectorizer(min_df=self.min_df_relative,
+                                                     ngram_range=(1,
+                                                                  self.ngram_range)).fit(X[feature])
+                                     for feature in
+                                     X.columns]
+            else:
+                raise KeyError()
+        else:
+            raise ValueError("Your text data is not encoded in a pandas.DataFrame\n"
+                             "Please make sure to use a pandas.DataFrame and ensure"
+                             "that the text features are encoded as strings.")
+        return self
+
+    def transform(self, X: PIPELINE_DATA_DTYPE) -> PIPELINE_DATA_DTYPE:
+        X_new = None
+        if self.preprocessor is None:
+            raise NotImplementedError()
+        # iterate over the pretrained preprocessors and columns and transform the data
+        for preprocessor, feature in zip(self.preprocessor, X.columns):
+            if X_new is None:
+                # possiblity to add TruncatedSVD here
+                X_new = preprocessor.transform(X[feature])
+            else:
+                X_new = hstack([X_new, preprocessor.transform(X[feature])])
+        return X_new
```

Comment:
@mfeurer 

So we text transform each column with it's individual preprocessor but still condense it all in to a single array as output?

I would have thought there is an array for each column?

Reply:
We could delete the TSVD as seperat pipeline step and implement it directly in the preprocessing.

Reply:
Could be part of #1373?